src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -118,6 +120,18 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		this._register(this._configurationService.onDidChangeConfiguration(e => {
 			this.doubleClickEditEnabled = this._configurationService.getValue('notebook.enableDoubleClickEdit');
 		}));
+		this._register(DOM.addDisposableListener(window, DOM.EventType.KEY_DOWN, (e: KeyboardEvent) => {
+			// Check that there are no active elements to prevent the execution of undo/redo commands in other elements.
+			// We need to make sure that there are no active cells to prevent triggering undo/redo within code cells and check that text cells are not in edit mode.
+			if (this.isActive() && !this.areElementsActive() && (this.activeCellId === '' || !this.cellToolbar.first.getEditCellAction().editMode)) {
```

Comment:
Is this logic necessary because it's capturing the key events before they get handled by the editors? Because I would expect that when those handle the undo/redo that they stop propagation already.

Anyways this is probably fine to go in for now, but I want to follow up at looking at this more. We want to be careful with how we handle key events since historically we've had some major issues when we deviate from how VS Code handles these things. If possible registering this stuff as actual commands with keybindings would be very much preferred (since that way we actually let the user control the keybindings as well...)

If you could look into how Notebooks are hooking up their undo/redo that'd be a good place to start.

Reply:
@barbaravaldez Could you comment on my first question? What happens if you remove this?

Reply:
The only logic that it's strictly necessary is `this.isActive() && this.activeCellId === ''` to prevent the undo/redo from happening in other notebooks and to prevent the execution of undo/redo in the cell.

I'll remove the other conditions.